packages/react-query/src/useBaseQuery.ts

```diff
@@ -77,10 +77,17 @@ export function useBaseQuery<
 
   useSyncExternalStore(
     React.useCallback(
-      (onStoreChange) =>
-        isRestoring
+      (onStoreChange) => {
+        const unsubscribe = isRestoring
           ? () => undefined
-          : observer.subscribe(notifyManager.batchCalls(onStoreChange)),
+          : observer.subscribe(notifyManager.batchCalls(onStoreChange))
+
+        // Update result to make sure we did not miss any query updates
+        // between creating the observer and subscribing to it.
+        observer.updateResult()
```

Comment:
Looks interesting, I'll try to test it in my project

Reply:
codesandbox preview seems to be having issues right now. You could patch-package the fix in your project, or we just ship it and trust the test coverage.

Reply:
The main problem, that I cannot patch) It's to long story to describe)

Let's ship it, cause looks like this fix might help.